File: src/main/java/org/dukcode/ps/codetree/trail02/chapter01/lesson02/ChallengeDeterminingTheWholeNumberUsingAFunction.java
package org.dukcode.ps.codetree.trail02.chapter01.lesson02;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

/**
 * @see <a
 * href="https://www.codetree.ai/ko/trails/complete/curated-cards/challenge-determining-the-whole-number-using-a-function/description">LINK</a>
 */
public class ChallengeDeterminingTheWholeNumberUsingAFunction {

  private static BufferedReader br;
  private static BufferedWriter bw;
  private static StringTokenizer st;

  public static void main(String[] args) throws IOException {
    br = new BufferedReader(new InputStreamReader(System.in));
    bw = new BufferedWriter(new OutputStreamWriter(System.out));

    st = new StringTokenizer(br.readLine());
    int frIn = Integer.parseInt(st.nextToken());
    int toIn = Integer.parseInt(st.nextToken());

    bw.write(String.valueOf(cntWholeNums(frIn, toIn)));

    br.close();
    bw.close();

  }

  private static int cntWholeNums(int frIn, int toIn) {
    int ret = 0;
    for (int num = frIn; num <= toIn; num++) {
      if (isWholeNum(num)) {
        ret++;
      }
    }
    return ret;
  }

  private static boolean isWholeNum(int num) {
    return num % 2 != 0 && num % 10 != 5 && (num % 3 != 0 || num % 9 == 0);
  }
}

File: src/main/kotlin/org/dukcode/ps/codetree/trail02/chapter01/lesson02/ChallengeDeterminingTheWholeNumberUsingAFunction.kt
package org.dukcode.ps.codetree.trail02.chapter01.lesson02

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.StringTokenizer

private val br = BufferedReader(InputStreamReader(System.`in`))
private val bw = BufferedWriter(OutputStreamWriter(System.out))

/**
 * @see <a href="https://www.codetree.ai/ko/trails/complete/curated-cards/challenge-determining-the-whole-number-using-a-function/description">LINK</a>
 */
fun main() {
    val st = StringTokenizer(br.readLine())
    val a = st.nextToken().toInt()
    val b = st.nextToken().toInt()

    bw.write(countWholeNumber(a, b).toString())

    bw.flush()
}

private fun countWholeNumber(
    frIn: Int,
    toIn: Int,
): Int =
    (frIn..toIn)
        .count { isWholeNumber(it) }

private fun isWholeNumber(num: Int): Boolean =
    num % 2 != 0 &&
        num % 10 != 5 &&
        (num % 3 != 0 || num % 9 == 0)
